src/fmi4c_utils.c

```diff
@@ -13,9 +13,10 @@ const char* getFunctionName(const char* modelName, const char* functionName) {
         return functionName;    //!< Do not change function name if model name is empty
     }
     char* fullName = (char*)malloc(strlen(modelName)+strlen(functionName)+2);
-    strcpy(fullName, modelName);
-    strcat(fullName, "_");
-    strcat(fullName, functionName);
+    strncpy(fullName, modelName, strlen(modelName)+strlen(functionName)+2);
```

Comment:
This still generates a warning for me, (about using the length of the source modelName) but obviously it is OK in this case since the destination is malloced to calculated size above.
I tried adding `const unsigned long maxLength = strlen(modelName)+strlen(functionName)+2;` and then reusing that variable in all of these lines (cleaner code). But it did not help, the compiler still knows that the size argument is based on the source length. So I guess we can let it be like this.